core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,401 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the contribution stats view.
+ */
+
+import { Component, ElementRef, HostListener, Input, ViewChild } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { ContributionAndReviewStatsService, QuestionContributionBackendDict, QuestionReviewBackendDict, TranslationContributionBackendDict, TranslationReviewBackendDict } from '../services/contribution-and-review-stats.service';
+import { UserService } from 'services/user.service';
+import { LanguageUtilService } from 'domain/utilities/language-util.service';
+import { AppConstants } from 'app.constants';
+
+interface Option {
+  contributionType: string;
+  displayName: string;
+}
+
+interface PageableStats {
+  language?: string;
+  currentPageStartIndex: number;
+  // This is the index of the last item of the stats data of a given page.
```

Comment:
```suggestion
  // Index (exclusive) of the last item of the current stats page.
```

The "exclusive" is the important part since it's not actually the index of the item, which can be confusing.

Reply:
Done

Reply:
I don't really understand what "exclusive" means here. Maybe adding an example to the comment could clarify? (Or maybe change the way it's represented so it's not "exclusive"?)

Reply:
We can just increment `currentPageStartIndex` and check the current page based on it. So I removed this field.